google-cloud-clients/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ModelId.java

```diff
@@ -69,12 +68,17 @@ private ModelId(String project, String dataset, String model) {
 
   /** Creates a model identity given project, dataset, and model identifiers. * */
   public static ModelId of(String project, String dataset, String model) {
-    return new ModelId(checkNotNull(project), checkNotNull(dataset), checkNotNull(model));
+    checkArgument(!isNullOrEmpty(project), "Provided project is null or empty");
+    checkArgument(!isNullOrEmpty(dataset), "Provided dataset is null or empty");
+    checkArgument(!isNullOrEmpty(model), "Provided model is null or empty");
+    return new ModelId(project, dataset, model);
   }
 
   /** Creates a model identity given dataset and model identifiers. * */
   public static ModelId of(String dataset, String model) {
-    return new ModelId(null, checkNotNull(dataset), checkNotNull(model));
+    checkArgument(!isNullOrEmpty(dataset), "Provided dataset is null or empty");
+    checkArgument(!isNullOrEmpty(model), "Provided model is null or empty");
+    return new ModelId(null, dataset, model);
```

Comment:
I think you can ignore these two checks here as there is already checks in underlying xor

Reply:
Can you add these checks in ModelId constructor, by doing that you have to only check in one place instead of these two